### PR TITLE
Publish event topic when registering a new user

### LINKED ISSUE #2584: Some UI pages crash if user credentials file can't be found
## Describe the bug
If after running the Agent, the user credentials file in the data directory can not be accessed for some reason, the security report page and the events page crash.

## To Reproduce
1. Run the Island and an Agent.
2. Stop the Island. Delete the user credentials file from the data directory. Run the Island again.
3. Go to the security report page or the events page and see it crash.

## Expected behavior
They shouldn't crash. Not finding the user credentials file should be handled better.

## Screenshots
https://user-images.githubusercontent.com/44770317/201935938-76937533-7950-458b-9ddd-59f57c7acd3d.mp4

## Tasks
- [ ] Fix it (0.5d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes #2584.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

https://user-images.githubusercontent.com/15820737/203600318-f0615cdc-aa00-413f-8f6f-ce674926c19d.mov

### CODE REVIEW

File: monkey/monkey_island/cc/services/authentication_service.py [hunk 1]
```diff
@@ -46,6 +49,12 @@ def register_new_user(self, username: str, password: str):
         if not username or not password:
             raise InvalidRegistrationCredentialsError("Username or password can not be empty.")
 
+        self._island_event_queue.publish(IslandEventTopic.CLEAR_SIMULATION_DATA)
+        self._island_event_queue.publish(IslandEventTopic.RESET_AGENT_CONFIGURATION)
+        self._island_event_queue.publish(
+            topic=IslandEventTopic.SET_ISLAND_MODE, mode=IslandMode.UNSET
+        )
+
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/services/authentication_service.py:57]: I was not sure about this. Thanks!

File: monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py [hunk 2]
```diff
@@ -39,56 +40,77 @@ def mock_repository_encryptor():
     return MagicMock(spec=ILockableEncryptor)
 
 
-def test_needs_registration__true(tmp_path, mock_repository_encryptor):
+@pytest.fixture
+def mock_island_event_queue():
+    return MagicMock(spec=IIslandEventQueue)
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py:45]: Just use `autouse` for this and the fixture above. There's no need to have to pass them to every test.

**Review by shreyamalviya — APPROVED** [review]
Approved with one change

File: monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py:45]: `autouse` doesn't work when the fixture returns an object. https://stackoverflow.com/a/37046403

File: monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py [hunk 3]
```diff
@@ -35,34 +36,47 @@ def get_user_credentials(self, username: str) -> UserCredentials:
 
 
 @pytest.fixture
-def mock_repository_encryptor():
+def mock_repository_encryptor(autouse=True):
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py:39]: You don't need to pass these to the functions if you're using autouse.

**Review by shreyamalviya — CHANGES_REQUESTED** [review]

File: monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/tests/unit_tests/monkey_island/cc/services/test_authentication_service.py:39]: When you return an object you need to pass them. I don't know how this even got here.

**Review by shreyamalviya — APPROVED** [review]

**Pull request merged**